ci-operator/step-registry/aws/deprovision/stacks/aws-deprovision-stacks-commands.sh

```
@@ -11,19 +11,170 @@ if [[ "${CLUSTER_TYPE:-}" =~ ^aws-s?c2s$ ]]; then
+                local vpc_id
+                vpc_id=$(aws --region $REGION ec2 describe-internet-gateways \
+                    --internet-gateway-ids "${physical_id}" \
+                    --query "InternetGateways[0].Attachments[0].VpcId" --output text 2>/dev/null)
+                if [[ -n "${vpc_id}" && "${vpc_id}" != "None" ]]; then
+                    aws --region $REGION ec2 detach-internet-gateway \
+                        --internet-gateway-id "${physical_id}" --vpc-id "${vpc_id}" 2>/dev/null || true
+                fi
+                aws --region $REGION ec2 delete-internet-gateway --internet-gateway-id "${physical_id}" 2>/dev/null || true
+                ;;
+            AWS::EC2::NatGateway)
+                aws --region $REGION ec2 delete-nat-gateway --nat-gateway-id "${physical_id}" 2>/dev/null || true
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)